gcloud-java-compute/src/test/java/com/google/gcloud/compute/AddressIdTest.java

```diff
@@ -0,0 +1,114 @@
+      "https://www.googleapis.com/compute/v1/projects/project/regions/region/addresses/addr";
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testOf() {
+    GlobalAddressId addressId = GlobalAddressId.of(PROJECT, NAME);
+    assertEquals(PROJECT, addressId.project());
+    assertEquals(NAME, addressId.address());
+    assertEquals(GLOBAL_URL, addressId.selfLink());
+    addressId = GlobalAddressId.of(NAME);
+    assertNull(addressId.project());
+    assertEquals(NAME, addressId.address());
+    RegionAddressId regionAddressId = RegionAddressId.of(PROJECT, REGION, NAME);
+    assertEquals(PROJECT, regionAddressId.project());
+    assertEquals(REGION, regionAddressId.region());
+    assertEquals(NAME, regionAddressId.address());
+    assertEquals(REGION_URL, regionAddressId.selfLink());
+    regionAddressId = RegionAddressId.of(RegionId.of(PROJECT, REGION), NAME);
+    assertEquals(PROJECT, regionAddressId.project());
+    assertEquals(REGION, regionAddressId.region());
+    assertEquals(NAME, regionAddressId.address());
+    assertEquals(REGION_URL, regionAddressId.selfLink());
+    regionAddressId = RegionAddressId.of(REGION, NAME);
+    assertNull(regionAddressId.project());
+    assertEquals(REGION, regionAddressId.region());
+    assertEquals(NAME, regionAddressId.address());
+  }
+
+  @Test
+  public void testToAndFromUrl() {
+    GlobalAddressId addressId = GlobalAddressId.of(PROJECT, NAME);
+    compareAddressId(addressId, GlobalAddressId.fromUrl(addressId.selfLink()));
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("notMatchingUrl is not a valid global address URL");
+    GlobalAddressId.fromUrl("notMatchingUrl");
+    RegionAddressId regionAddressId = RegionAddressId.of(PROJECT, REGION, NAME);
+    compareRegionAddressId(regionAddressId, RegionAddressId.fromUrl(regionAddressId.selfLink()));
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("notMatchingUrl is not a valid global address URL");
+    RegionAddressId.fromUrl("notMatchingUrl");
+  }
+
+  @Test
+  public void testSetProjectId() {
+    GlobalAddressId addressId = GlobalAddressId.of(PROJECT, NAME);
+    assertSame(addressId, addressId.setProjectId(PROJECT));
+    compareAddressId(addressId, GlobalAddressId.of(NAME).setProjectId(PROJECT));
+    RegionAddressId regionAddressId = RegionAddressId.of(PROJECT, REGION, NAME);
+    compareRegionAddressId(regionAddressId, RegionAddressId.of(REGION, NAME).setProjectId(PROJECT));
+  }
+
+  @Test
+  public void testMatchesUrl() {
+    assertTrue(GlobalAddressId.matchesUrl(GlobalAddressId.of(PROJECT, NAME).selfLink()));
+    assertFalse(GlobalAddressId.matchesUrl("notMatchingUrl"));
+    assertTrue(RegionAddressId.matchesUrl(RegionAddressId.of(PROJECT, REGION, NAME).selfLink()));
+    assertFalse(RegionAddressId.matchesUrl("notMatchingUrl"));
+  }
+
+  private void compareAddressId(GlobalAddressId expected, GlobalAddressId value) {
```

Comment:
I am not sure this is the purpose of the method. Even though such an assert would always pass the purpose of `compareXXX` methods is to compare two objects, not to assert their content. This should be already covered in `testOf` methods.